Can't run worker with container in multiworker setup if container worker is not the entry worker

### What versions & operating system are you using?

  System:
    OS: Linux 5.15 Ubuntu 24.04.1 LTS 24.04.1 LTS (Noble Numbat)
    CPU: (8) x64 Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz
    Memory: 4.91 GB / 7.63 GB
    Container: Yes
    Shell: 5.2.21 - /bin/bash
  Binaries:
    Node: 22.21.1 - /home/nzakas/.nvm/versions/node/v22.21.1/bin/node
    npm: 10.9.4 - /home/nzakas/.nvm/versions/node/v22.21.1/bin/npm
  npmPackages:
    wrangler: ^4.54.0 => 4.54.0 

### Please provide a link to a minimal reproduction

https://github.com/nzakas

### Describe the Bug

1. Create a worker without a container (`example`)
2. Create a worker with a container (`container`) -- the container can do anything, it doesn't matter
3. Try to run both together `npx wrangler dev -c example/wrangler.jsonc container/wrangler.jsonc`

Running `npx wrangler dev -c container/wrangler.jsonc` works fine. 

(Apologies for the lack of a repro, hopefully this description is easy to follow.)

### Please provide any relevant error logs

```
--- 2025-12-13T19:54:15.749Z log
*** Fatal uncaught kj::Exception: kj/table.c++:49: failed: inserted row already exists in table
---

--- 2025-12-13T19:54:15.749Z log
stack: /home/nzakas/projects/savelinks/node_modules/@cloudflare/workerd-linux-64/bin/workerd@4ed519f /home/nzakas/projects/savelinks/node_modules/@cloudflare/workerd-linux-64/bin/workerd@1fc4a40 /home/nzakas/projects/savelinks/node_modules/@cloudflare/workerd-linux-64/bin/workerd@1fc4934 /home/nzakas/projects/savelinks/node_modules/@cloudflare/workerd-linux-64/bin/workerd@1f9f5e8 /home/nzakas/projects/savelinks/node_modules/@cloudflare/workerd-linux-64/bin/workerd@1f5593d /home/nzakas/projects/savelinks/node_modules/@cloudflare/workerd-linux-64/bin/workerd@1fbead6 /home/nzakas/projects/savelinks/node_modules/@cloudflare/workerd-linux-64/bin/workerd@1fc1959 /home/nzakas/projects/savelinks/node_modules/@cloudflare/workerd-linux-64/bin/workerd@1f2db7e /home/nzakas/projects/savelinks/node_modules/@cloudflare/workerd-linux-64/bin/workerd@4ec8638 /home/nzakas/projects/savelinks/node_modules/@cloudflare/workerd-linux-64/bin/workerd@4ec8a80 /home/nzakas/projects/savelinks/node_modules/@cloudflare/workerd-linux-64/bin/workerd@4ec6935 /home/nzakas/projects/savelinks/node_modules/@cloudflare/workerd-linux-64/bin/workerd@4ec6748 /home/nzakas/projects/savelinks/node_modules/@cloudflare/workerd-linux-64/bin/workerd@1f17c75 /lib/x86_64-linux-gnu/libc.so.6@2a1c9 /lib/x86_64-linux-gnu/libc.so.6@2a28a /home/nzakas/projects/savelinks/node_modules/@cloudflare/workerd-linux-64/bin/workerd@1f17024
---

--- 2025-12-13T19:54:15.751Z debug
Error in MultiworkerRuntimeController: Error reloading local server
 MiniflareCoreError [ERR_RUNTIME_FAILURE]: The Workers runtime failed to start. There is likely additional logging output above.
    at #assembleAndUpdateConfig (/home/nzakas/projects/savelinks/node_modules/miniflare/dist/src/index.js:62438:13)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async Mutex.runWith (/home/nzakas/projects/savelinks/node_modules/miniflare/dist/src/index.js:38902:48)
    at async #waitForReady (/home/nzakas/projects/savelinks/node_modules/miniflare/dist/src/index.js:62525:5)
    at async #onBundleComplete (/home/nzakas/projects/savelinks/node_modules/wrangler/wrangler-dist/cli.js:284191:35)
    at async Mutex.runWith (/home/nzakas/projects/savelinks/node_modules/miniflare/dist/src/index.js:38902:48) {
  code: 'ERR_RUNTIME_FAILURE',
  cause: undefined
}
---

--- 2025-12-13T19:54:15.751Z debug
=> Error contextual data: undefined
---

--- 2025-12-13T19:54:15.752Z log

---

--- 2025-12-13T19:54:15.826Z error
✘ [ERROR] The Workers runtime failed to start. There is likely additional logging output above.


---

--- 2025-12-13T19:54:15.827Z debug
MiniflareCoreError [ERR_RUNTIME_FAILURE]: The Workers runtime failed to start. There is likely additional logging output above.
    at #assembleAndUpdateConfig (/home/nzakas/projects/savelinks/node_modules/miniflare/dist/src/index.js:62438:13)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async Mutex.runWith (/home/nzakas/projects/savelinks/node_modules/miniflare/dist/src/index.js:38902:48)
    at async #waitForReady (/home/nzakas/projects/savelinks/node_modules/miniflare/dist/src/index.js:62525:5)
    at async #onBundleComplete (/home/nzakas/projects/savelinks/node_modules/wrangler/wrangler-dist/cli.js:284191:35)
    at async Mutex.runWith (/home/nzakas/projects/savelinks/node_modules/miniflare/dist/src/index.js:38902:48)
---

--- 2025-12-13T19:54:15.827Z log
If you think this is a bug then please create an issue at https://github.com/cloudflare/workers-sdk/issues/new/choose
---

--- 2025-12-13T19:54:22.673Z debug
Sentry: Capturing exception MiniflareCoreError [ERR_RUNTIME_FAILURE]: The Workers runtime failed to start. There is likely additional logging output above.
---

```

--- a/.changeset/fix-browser-rendering-multi-worker.md
+++ b/.changeset/fix-browser-rendering-multi-worker.md
@@ -1,0 +1,7 @@
+---
+"miniflare": patch
+---
+
+fix: allow multiple workers with browser bindings in dev
+
+You can now run multiple workers with multiple browser bindings in miniflare. Previously, this would crash with `kj/table.c++:49: failed: inserted row already exists in table`.

--- a/packages/miniflare/src/plugins/browser-rendering/index.ts
+++ b/packages/miniflare/src/plugins/browser-rendering/index.ts
@@ -54,7 +54,7 @@ export const BROWSER_RENDERING_PLUGIN: Plugin<
 				service: {
 					name: getUserBindingServiceName(
 						BROWSER_RENDERING_PLUGIN_NAME,
-						options.browserRendering.binding,
+						"service",
 						options.browserRendering.remoteProxyConnectionString
 					),
 				},
@@ -78,7 +78,7 @@ export const BROWSER_RENDERING_PLUGIN: Plugin<
 			{
 				name: getUserBindingServiceName(
 					BROWSER_RENDERING_PLUGIN_NAME,
-					options.browserRendering.binding,
+					"service",
 					options.browserRendering.remoteProxyConnectionString
 				),
 				worker: options.browserRendering.remoteProxyConnectionString
@@ -107,6 +107,7 @@ export const BROWSER_RENDERING_PLUGIN: Plugin<
 							durableObjectNamespaces: [
 								{
 									className: "BrowserSession",
+									uniqueKey: "miniflare-BrowserSession",
 								},
 							],
 							durableObjectStorage: { inMemory: kVoid },

--- a/packages/miniflare/test/plugins/browser/index.spec.ts
+++ b/packages/miniflare/test/plugins/browser/index.spec.ts
@@ -96,15 +96,15 @@ describe.sequential("browser rendering", { timeout: 20_000 }, () => {
 		{ retry: 3 },
 		async ({ expect }) => {
 			const workerScript = (bindingName: string) => `
-export default {
-	async fetch(request, env) {
-		if (request.url.endsWith("session")) {
-			const newBrowserSession = await env.${bindingName}.fetch("https://localhost/v1/acquire")
-			return new Response(await newBrowserSession.text())
-		}
-	}
-};
-`;
+			export default {
+				async fetch(request, env) {
+					if (request.url.endsWith("session")) {
+						const newBrowserSession = await env.${bindingName}.fetch("https://localhost/v1/acquire")
+						return new Response(await newBrowserSession.text())
+					}
+				}
+			};
+			`;
 			const mf = new Miniflare({
 				workers: [
 					{

--- a/packages/miniflare/test/plugins/browser/index.spec.ts
+++ b/packages/miniflare/test/plugins/browser/index.spec.ts
@@ -91,6 +91,46 @@ describe.sequential("browser rendering", { timeout: 20_000 }, () => {
 		expect(text.includes("sessionId")).toBe(true);
 	});
 
+	test(
+		"two workers with different browser bindings can coexist",
+		{ retry: 3 },
+		async ({ expect }) => {
+			const workerScript = (bindingName: string) => `
+export default {
+	async fetch(request, env) {
+		if (request.url.endsWith("session")) {
+			const newBrowserSession = await env.${bindingName}.fetch("https://localhost/v1/acquire")
+			return new Response(await newBrowserSession.text())
+		}
+	}
+};
+`;
+			const mf = new Miniflare({
+				workers: [
+					{
+						name: "worker-a",
+						compatibilityDate: "2024-11-20",
+						modules: true,
+						script: workerScript("BROWSER_A"),
+						browserRendering: { binding: "BROWSER_A" },
+					},
+					{
+						name: "worker-b",
+						compatibilityDate: "2024-11-20",
+						modules: true,
+						script: workerScript("BROWSER_B"),
+						browserRendering: { binding: "BROWSER_B" },
+					},
+				],
+			});
+			useDispose(mf);
+
+			const res = await mf.dispatchFetch("https://localhost/session");
+			const text = await res.text();
+			expect(text.includes("sessionId")).toBe(true);
+		}
+	);
+
 	const BROWSER_WORKER_CLOSE_SCRIPT = `
 ${sendMessage.toString()}
 ${waitForClosedConnection.toString()}

